### PR TITLE
Use in memory config in cli tests

### PR DESCRIPTION
**Pull request by tomprince**
This is preparation for writing the CLI test for #210.

### CODE REVIEW

File: src/magic_folder/list.py [hunk 1]
```diff
@@ -17,19 +17,14 @@
...
 @inlineCallbacks
-def magic_folder_list(reactor, config, http_client, output, as_json=False, include_secret_information=False):
+def magic_folder_list(reactor, config, client, output, as_json=False, include_secret_information=False):
     """
     List folders associated with a node.
 
     :param GlobalConfigDatabase config: our configuration
 
-    :param HTTPClient http_client: a treq.HTTPClient instance we can
+    :param MagicFolderClient client: a treq.HTTPClient instance we can
```
Comment (meejah) [inline src/magic_folder/list.py:27]: comment seems wrong now?

File: src/magic_folder/test/cli/test_magic_folder.py [hunk 2]
```diff
@@ -533,22 +526,65 @@ def test_config_directory_is_file(self):
...
+                api_endpoint=Equals(u"tcp:5555"),
+                tahoe_node_directory=Equals(nodedir.path),
+            )
+        )
+
+    @inline_callbacks
+    def test_default_config_directory(self):
+        """
+        Not passing a --config loads the configuration from the default directory.
+        """
```
Comment (meejah) [inline src/magic_folder/test/cli/test_magic_folder.py:571]: This test doesn't seem to match what it says here; how is this testing the default config directory?

**Review by meejah — APPROVED** [review]
two things inline to address

File: src/magic_folder/test/cli/test_magic_folder.py (hunk 2 above)
Comment (tomprince) [reply src/magic_folder/test/cli/test_magic_folder.py:571]: I've added a comment explaining how we are testing the default.

**Pull request merged**